.github/workflows/bump_project_version.yml

```diff
@@ -0,0 +1,42 @@
+#############################################################################
+# GitHub Action to Bump Project Version.
+#
+# Workflow starts every day at the end of the day.
+#
+#############################################################################
+name: "Bump CS Version in REPO"
+on:
+    schedule:
+        - cron: '0 0 * * *'
+    workflow_dispatch:
+permissions:
+    contents: write
+    pull-requests: write
+jobs:
+    bump:
+        name: Bump CS version in REPO
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout the latest code
+              uses: actions/checkout@v2
+              with:
+                  token: ${{ secrets.GITHUB_TOKEN }}
+            - name: Get Version
+              id: VERSION
+              run: |
+                  MAVEN_METADATA=https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml
+                  LATEST_VERSION=$(curl $MAVEN_METADATA | grep -oPm1 "(?<=<latest>)[^<]+")
+                  echo "::set-output name=LATEST_VERSION::$LATEST_VERSION"
+            - name: Run Shell Script
+              run: |
+                  ./.ci/bump-cs-version-in-patch-diff-report-tool.sh ${{ steps.VERSION.outputs.LATEST_VERSION }}
+                  ./.ci/bump-cs-version-in-releasenotes-builder.sh ${{ steps.VERSION.outputs.LATEST_VERSION }}
+            - name: Check Changes
+              run: |
+                  git diff
+            - name: Commit and Push Changes
+              run: |
+                  git config --global user.name 'github-actions[bot]'
+                  git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+                  git commit -am "minor: Bump project version to ${{ steps.VERSION.outputs.LATEST_VERSION }}"
```

Comment:
you need to protect creation of commit without changes, this line will  fail if if changes detected.
https://github.com/checkstyle/checkstyle/blob/a97c3e8196c90fd8cf90a755f0b7ef5fd2824d29/.ci/validation.sh#L1008-L1010

Reply:
Changes Done